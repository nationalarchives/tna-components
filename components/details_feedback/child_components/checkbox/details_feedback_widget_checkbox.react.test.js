import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './details_feedback_widget_checkbox.react';

//let testRenderer, testRendererToJson;

// Snapshot test 1
it('renders the checkbox component correctly', () => {
  const items = ['Hello world', 'this-is-the-id'];
  const tree = renderer
    .create(<Checkbox labelText={items[0]} id={items[1]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Snapshot test 2
it('renders the checkbox component with optional props', () => {
  const items = [
    'Hello world',
    'this-is-the-id',
    () => console.log('Function test')
  ];
  const tree = renderer
    .create(
      <Checkbox
        labelText={items[0]}
        id={items[1]}
        onClick={items[2]}
        autoFocus="autofocus"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
