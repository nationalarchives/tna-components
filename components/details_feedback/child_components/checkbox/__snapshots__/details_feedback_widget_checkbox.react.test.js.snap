// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the checkbox component correctly 1`] = `
<div
  className="checkbox"
>
  <input
    id="this-is-the-id"
    type="checkbox"
    value="this-is-the-id"
  />
  <label
    for="this-is-the-id"
  >
    Hello world
  </label>
</div>
`;

exports[`renders the checkbox component with optional props 1`] = `
<div
  className="checkbox"
>
  <input
    autoFocus="autofocus"
    id="this-is-the-id"
    onClick={[Function]}
    type="checkbox"
    value="this-is-the-id"
  />
  <label
    for="this-is-the-id"
  >
    Hello world
  </label>
</div>
`;
